server/db_mysql.go

```diff
@@ -129,6 +131,122 @@ func (ms *MsqlCache) NextFixity(cutoff time.Time) string {
+                return nil
+        }
+
+        // Handle for nil time value
+	if thisWhen.Valid {
+	    thisFixity.Scheduled_time = thisWhen.Time
+	} else {
+	    thisFixity.Scheduled_time = time.Time{}
+	}
+
+        log.Println("id= ", thisFixity.Id, "scheduled_time= ", thisFixity.Scheduled_time, "status= ", thisFixity.Status, "notes= ", thisFixity.Notes)
+        return thisFixity
+}
+
+// GetFixity
+func (ms *MsqlCache) GetFixity( start string, end string, item string, status string)  []*fixity {
+        query := buildQuery(start, end, item, status)
+	log.Println("GET /fixity query= ", query)
+	var thisWhen mysql.NullTime
+        var fixityResults []*fixity
+
+        rows, err := ms.db.Query(query)
+        if err == sql.ErrNoRows {
+                // no next record
+                return nil
+        } else if err != nil {
+                log.Println("GetFixity Query MySQL", err.Error())
+                return nil
+        }
+        defer rows.Close()
+
+	for rows.Next() {
+		var thisFixity = new(fixity)
+		scanErr := rows.Scan(&thisFixity.Id, &thisFixity.Item, &thisWhen, &thisFixity.Status, &thisFixity.Notes)
+		if scanErr != nil {
+		    log.Println("GetFixity Scan MySQL", err.Error())
+                    return nil
+		}
+        	// Handle for nil time value
+		if thisWhen.Valid {
+	    		thisFixity.Scheduled_time = thisWhen.Time
+		} else {
+	    		thisFixity.Scheduled_time = time.Time{}
+		}
+                log.Println("id= ", thisFixity.Id, "item =", thisFixity.Item, "scheduled_time", thisFixity.Scheduled_time, "status= ", thisFixity.Status, "notes= ", thisFixity.Notes)
+		fixityResults = append(fixityResults,thisFixity)
+        }
+
+        return fixityResults
+}
+
+// construct an return an sql query, using the parameters passed
+func buildQuery( start string, end string, item string, status string) string {
+	var query bytes.Buffer
+	query.WriteString("SELECT  id, item, scheduled_time, status, notes FROM fixity")
+
+	params := []string{ "start", "end", "item", "status"}
+
+	conjunction := " WHERE "
+
+	for _, param := range params {
+
+		switch param {
+			case "start":
+				if start != "*" {
+					startQuery := []string{ conjunction, "scheduled_time >= '", start, "'"}
+					query.WriteString(strings.Join(startQuery,""))
+				} else {
+					continue
+				}
+			case "end":
+				if end != "*" {
+					endQuery := []string{ conjunction, "scheduled_time <= '", end, "'"}
+					query.WriteString(strings.Join(endQuery,""))
+				} else {
+					continue
+				}
+			case "item":
+				if item != "" {
+					itemQuery := []string{conjunction,"item = '", item, "'"}
+					query.WriteString(strings.Join(itemQuery,""))
+				} else {
+					continue
+				}
+			case "status":
+				if status != "" {
+					statusQuery := []string{conjunction,"status = '", status, "'"}
+					query.WriteString(strings.Join(statusQuery,""))
+				} else {
+					continue
+				}
+		}
+
+		conjunction = " AND "
+	}
+
+	return query.String()
+}
```

Comment:
The design really requires a lot of engineering!

I would rather if the parameters were substituted into the query using the mysql query placeholders...but then how to balance that against having a crazy number of select statements.

Maybe make a statement using loop like above but use named placeholders (like https://golang.org/pkg/database/sql/#Named) and then supply all the named arguments in a Query() statement? If a certain select statement does not need an argument, no big deal...(unsure if this will work).